### ISSUE TITLE
why train for 3 times

### ISSUE THREAD

**Issue by xavier-xiadia** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#95]
hi，thanks for u work!  i have some questions. firstly, i tried trained from mobilev1 which get ap 0.38. i trained from previous model but only get 0.37 with -weight-only and 1 refinement stages. then i trained from the 0.38 model skipping 2nd step and get 0.392 with 2 refinement stages.  though it still cannot reach your scores, i wonder what is the meaning of the 2nd training.

**Comment by Daniil-Osokin** [comment]
Hi! It was done to speed up experiments. Model with one refinement stage trains faster, than full model with 5 refinement stages. In general it is ok to train the full model form MobileNet weights.

**Comment by xavier-xiadia** [comment]
@Daniil-Osokin Thank u for u reply! i eval the 2 refinement stage in 2500 pics, result is:
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.404
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.666
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.404
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.336
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.510
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.459
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.695
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.466
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.357
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.612
couldn't reach ' 2-stage network, retrained with all refinement stages 42.8 n/a 9' proposed in u paper.
the val dataset for the reason is 5000? or something wrong with my experiment?
my param:
batch-size 80
base-lr 4e-5
num-refinement-stages 2
iter num 130000
GPU v100*8

**Comment by Daniil-Osokin** [comment]
My congratulations! Actually you have reached the declared accuracy for this code, check the [README](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch#real-time-2d-multi-person-pose-estimation-on-cpu-lightweight-openpose). We have developed this work, when the mainstream DL framework was [Caffe](https://github.com/BVLC/caffe), so we used it to reach the results from the paper. And with PyTorch implementation we have obtained ~40 AP, so your result is similar to the [shared model](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch#pre-trained-model) and up to date.

**Comment by xavier-xiadia** [comment]
@Daniil-Osokin o~,thank u, i got it!

**Comment by Daniil-Osokin** [comment]
Ok, closing.

**Comment by MoonBunnyZZZ** [comment]
> @Daniil-Osokin Thank u for u reply! i eval the 2 refinement stage in 2500 pics, result is:
> Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets= 20 ] = 0.404
> Average Precision (AP) @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.666
> Average Precision (AP) @[ IoU=0.75 | area= all | maxDets= 20 ] = 0.404
> Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.336
> Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.510
> Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 20 ] = 0.459
> Average Recall (AR) @[ IoU=0.50 | area= all | maxDets= 20 ] = 0.695
> Average Recall (AR) @[ IoU=0.75 | area= all | maxDets= 20 ] = 0.466
> Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.357
> Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.612
> couldn't reach ' 2-stage network, retrained with all refinement stages 42.8 n/a 9' proposed in u paper.
> the val dataset for the reason is 5000? or something wrong with my experiment?
> my param:
> batch-size 80
> base-lr 4e-5
> num-refinement-stages 2
> iter num 130000
> GPU v100*8

@xavier-xiadia Hi, 2500 pics you used for validation are randomly chosen?

@Daniil-Osokin 
1)In Readme,it is said that the model  has 40% of AP on COCO validation set.This set is coco validation set  which have 5647 images? 
2)I used the pretained model of iter 370000 on coco validation set and just got AP
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.103
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.164
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.107
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.131
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.126
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 20 ] = 0.428
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets= 20 ] = 0.616
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets= 20 ] = 0.452
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets= 20 ] = 0.296
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets= 20 ] = 0.605
At first, I think there are something wrong in my code.But, the pretrained model have AP 0.386 on  val
subset. So the val code are ok.Could you tell me why?

**Comment by xavier-xiadia** [comment]
@MoonBunnyZZZ 2500pics are made from make_val_subset.py, it's random, the result of your training maybe wrong. you can check batch-size base-lr ,the training dataset and log the loss（if loss is decrease）

**Comment by MoonBunnyZZZ** [comment]
> @MoonBunnyZZZ 2500pics are made from make_val_subset.py, it's random, the result of your training maybe wrong. you can check batch-size base-lr ,the training dataset and log the loss（if loss is decrease）

@xavier-xiadia Thank you for your reply
1 Okay,I am clear about how to create 2500 pics
2 I used model file checkpoint_iter_370000.pth, which was trained by the @Daniil-Osokin.
   What confuse me is that the same model and the same  evaluation code on two val set(250 randomly picked and whole coco val set) have very different  AP result. 0.386 vs 0.103

**Comment by xavier-xiadia** [comment]
@MoonBunnyZZZ I didn't use checkpoint_iter_370000.pth, maybe u can train a new one by yourself to git rid of overfitting. i don't know how the model file was trained. Maybe something wrong with your COCO val?

**Comment by MoonBunnyZZZ** [comment]
> @MoonBunnyZZZ I didn't use checkpoint_iter_370000.pth, maybe u can train a new one by yourself to git rid of overfitting. i don't know how the model file was trained. Maybe something wrong with your COCO val?

Thank you!
I will check it again.
